Spigot-Server-Patches/0704-Custom-Advancement-Triggers.patch

```diff
@@ -0,0 +1,275 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 21 Apr 2021 23:55:48 -0700
+Subject: [PATCH] Custom Advancement Triggers
+
+
+diff --git a/src/main/java/io/papermc/paper/advancements/CraftAdvancementManager.java b/src/main/java/io/papermc/paper/advancements/CraftAdvancementManager.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..e20b6773bd6e6a46e775732ebf9d8d48e5bc94dd
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/advancements/CraftAdvancementManager.java
+@@ -0,0 +1,29 @@
++package io.papermc.paper.advancements;
++
++import com.google.common.collect.Maps;
++import net.minecraft.advancements.CriterionTriggers;
++import org.bukkit.NamespacedKey;
++
++import java.util.Map;
++
++public class CraftAdvancementManager implements AdvancementManager {
++
++    private final Map<NamespacedKey, AdvancementTrigger> CUSTOM_TRIGGERS = Maps.newHashMap();
```

Comment:
Why does this variable use the constant naming convention ? While it is a `final` variable, this does not qualify as a constant and should hence follow the default java variable/field naming convention.

Reply:
Yep, that was left over from when it was a static field and I just didn’t change it.